tests/testDipoleFitter.py

```diff
@@ -0,0 +1,331 @@
+
+            self.assertClose((result.psfFitPosFlux + abs(result.psfFitNegFlux))/2.,
+                             params.flux[i], rtol=rtol)
+            self.assertClose(result.psfFitPosCentroidX, params.xc[i] + offsets[i], rtol=rtol)
+            self.assertClose(result.psfFitPosCentroidY, params.yc[i] + offsets[i], rtol=rtol)
+            self.assertClose(result.psfFitNegCentroidX, params.xc[i] - offsets[i], rtol=rtol)
+            self.assertClose(result.psfFitNegCentroidY, params.yc[i] - offsets[i], rtol=rtol)
+
+    def _runDetection(self, params):
+        """Run 'diaSource' detection on the diffim, including merging of
+        positive and negative sources.
+
+        Then run DipoleFitTask on the image and return the resulting catalog.
+        """
+
+        # Create the various tasks and schema -- avoid code reuse.
+        testImage = params.testImage
+        detectTask, schema = testImage.detectDipoleSources(doMerge=False)
+
+        measureConfig = measBase.SingleFrameMeasurementConfig()
+
+        measureConfig.slots.calibFlux = None
+        measureConfig.slots.modelFlux = None
+        measureConfig.slots.instFlux = None
+        measureConfig.slots.shape = None
+        measureConfig.slots.centroid = "ip_diffim_NaiveDipoleCentroid"
+        measureConfig.doReplaceWithNoise = False
+
+        measureConfig.plugins.names = ["base_CircularApertureFlux",
+                                       "base_PixelFlags",
+                                       "base_SkyCoord",
+                                       "base_PsfFlux",
+                                       "ip_diffim_NaiveDipoleCentroid",
+                                       "ip_diffim_NaiveDipoleFlux",
+                                       "ip_diffim_PsfDipoleFlux"]
+
+        # Disable aperture correction, which requires having an ApCorrMap attached to
+        # the Exposure (it'll warn if it's not present and we don't explicitly disable it).
+        measureConfig.doApplyApCorr = "no"
+
+        # Here is where we make the dipole fitting task. It can run the other measurements as well.
+        # This is an example of how to pass it a custom config.
+        dpFitPluginConfig = DipoleFitPluginConfig()
+        dpFitPluginConfig.verbose = params.verbose
+        measureTask = DipoleFitTask(config=measureConfig, schema=schema, dpFitPluginConfig=dpFitPluginConfig)
+
+        table = afwTable.SourceTable.make(schema)
+        detectResult = detectTask.run(table, testImage.diffim)
+        # catalog = detectResult.sources
+        # deblendTask.run(self.dipole, catalog, psf=self.dipole.getPsf())
+
+        fpSet = detectResult.fpSets.positive
+        fpSet.merge(detectResult.fpSets.negative, 2, 2, False)
+        sources = afwTable.SourceCatalog(table)
+        fpSet.makeSources(sources)
+
+        measureTask.run(sources, testImage.diffim, testImage.posImage, testImage.negImage)
+        return sources
+
+    def _checkTaskOutput(self, params, sources, rtol=None):
```

Comment:
It uses the "default" `rtol` from the `TestDipoleImage`. I will pass on `atol` for now, since the appropriate value will depend upon what is being tested (fluxes or coordinates).